app/controllers/api/functions.php

```diff
@@ -1926,9 +1951,38 @@
         $execution->setAttribute('responseBody', $executionResponse['body'] ?? '');
         $execution->setAttribute('responseHeaders', $headers);
 
-        $response
-            ->setStatusCode(Response::STATUS_CODE_CREATED)
-            ->dynamic($execution, Response::MODEL_EXECUTION);
+        $acceptTypes = \explode(', ', $request->getHeader('accept', 'multipart/form-data'));
+        $isJson = false;
+
+        foreach ($acceptTypes as $acceptType) {
+            if (\str_starts_with($acceptType, 'application/json') || \str_starts_with($acceptType, 'application/*')) {
+                $isJson = true;
+                break;
+            }
+        }
+
+        if ($isJson) {
+            $executionString = \json_encode($execution, JSON_UNESCAPED_UNICODE);
+            if (!$executionString) {
+                throw new Exception('Execution resulted in binary response, but JSON response does not allow binaries. Use "Accept: multipart/form-data" header to support binaries.', 400);
+            }
+
+            $response
+                ->setStatusCode(Response::STATUS_CODE_OK)
+                ->addHeader('content-type', 'application/json')
+                ->send($executionString);
```

Comment:
We cant do this, because response filters don't apply anymore. We need to keep using response filters.